src/benchmarks/real-world/ImageSharp/Codecs/EncodeGif.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.IO;
+using BenchmarkDotNet.Attributes;
+using SixLabors.ImageSharp.Formats.Gif;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using SixLabors.ImageSharp.Processing.Processors.Quantization;
+using SixLabors.ImageSharp.Tests;
+
+namespace SixLabors.ImageSharp.Benchmarks.Codecs
+{
+    
+    public class EncodeGif
+    {
+        // System.Drawing needs this.
+        private Stream bmpStream;
+        private Image<Rgba32> bmpCore;
+
+        // Try to get as close to System.Drawing's output as possible
+        private readonly GifEncoder encoder = new GifEncoder
+        {
+            Quantizer = new WebSafePaletteQuantizer(new QuantizerOptions { Dither = KnownDitherings.Bayer4x4 })
+        };
+
+        [Params(TestImages.Bmp.Car, TestImages.Png.Rgb48Bpp)]
+        public string TestImage { get; set; }
+
+        [GlobalSetup]
+        public void ReadImages()
+        {
+            if (this.bmpStream == null)
+            {
+                this.bmpStream = File.OpenRead(Path.Combine(TestEnvironment.InputImagesDirectoryFullPath, this.TestImage));
+                this.bmpCore = Image.Load<Rgba32>(this.bmpStream);
+                this.bmpStream.Position = 0;
+            }
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            this.bmpStream.Dispose();
+            this.bmpStream = null;
+            this.bmpCore.Dispose();
+        }
+
+        [Benchmark(Baseline = true, Description = "System.Drawing Gif")]
+        public void GifSystemDrawing()
+        {
+            using var memoryStream = new MemoryStream();
+        }
+
+        [Benchmark(Description = "ImageSharp Gif")]
+        public void GifImageSharp()
+        {
+            using var memoryStream = new MemoryStream();
```

Comment:
Same `memoryStream` issue as with `EncodeBmp`.

/cc @JimBobSquarePants we should also cleanup these.

Reply:
See my comments above.